src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F

```diff
@@ -829,6 +1303,70 @@ subroutine init_sfclayer(configs)
     case("sf_mynn")
        call sf_mynn_init(errmsg,errflg)
 
+    case("sf_myj") 
+       ! only lowlyr, ust, z0 inout, rest in only
+       if (.not.allocated(lowlyr_p)) allocate(lowlyr_p(ims:ime,jms:jme))
+       if (.not.allocated(ust_p)   ) allocate(ust_p(ims:ime,jms:jme)   )
+       if (.not.allocated(z0_p)    ) allocate(z0_p(ims:ime,jms:jme)    )
+       if (.not.allocated(xland_p) ) allocate(xland_p(ims:ime,jms:jme) )
+       if (.not.allocated(xice_p)  ) allocate(xice_p(ims:ime,jms:jme)  )
+       if (.not.allocated(ivgtyp_p)) allocate(ivgtyp_p(ims:ime,jms:jme))
+   
+       do j = jts,jte
+       do i = its,ite
+          !lowlyr_p(i,j) = lowlyr(i)
+          !ust_p(i,j)    = ust(i)
+          !z0_p(i,j)     = z0(i)
+          xland_p(i,j)  = xland(i)
+          xice_p(i,j)   = xice(i)
+          ivgtyp_p(i,j) = ivgtyp(i)
+       enddo
+       enddo
+
+       call MYJSFCINIT(lowlyr_p, ust_p, z0_p, xland_p, xice_p,        & 
+                       ivgtyp_p, config_do_restart, allowed_to_read,   &
+                       ids, ide, jds, jde, kds, kde, ims, ime, jms,    &
+                       jme, kms, kme, its, ite, jts, jte, kts, kte)
+       do j = jts,jte
+       do i = its,ite
+          lowlyr(i) = lowlyr_p(i,j)
+       !   ust(i)    = ust_p(i,j)
+       !   z0(i)    = z0_p(i,j)
+       enddo
+       enddo
+
```

Comment:
the block of code above this comment looks identical to the code below it. I think one of them should be removed.